### PR TITLE
Papal stuff and hats on wigs

### PR DESCRIPTION
**Pull request by WoolyAypa**
## About The Pull Request
- Papal clothing to the chaplain armament beacon
- Papal staff to the nullrod variants

- Some hats can now be put over wigs

## Changelog

:cl:
add: Adds papal clothing to the chaplain armament beacon
add: Adds a papal staff to the nullrod variants
tweak: Some hats can now be put over wigs
/:cl:

### CODE REVIEW

**Comment by ReturnToZender** [comment]
> ## About The Pull Request
> * Papal clothing to the chaplain armament beacon
> * Papal staff to the nullrod variants
> * Some hats can now be put over wigs
> 
> ## Changelog
> 🆑 add: Adds papal clothing to the chaplain armament beacon add: Adds a papal staff to the nullrod variants tweak: Some hats can now be put over wigs /🆑

> Error: /obj/item/nullrod/papal_staff using invalid icon_state, "papal_staff" in 'icons/mob/clothing/belt.dmi'
>  	FAILURE #1: /obj/item/nullrod/papal_staff using invalid icon_state, "papal_staff" in 'icons/mob/clothing/belt.dmi' 
> at code/modules/unit_tests/worn_icons.dm:116
  cries

**Comment by WoolyAypa** [comment]
bruhh...

**Comment by Cyprex** [comment]
Can be testmerged. I need to look into a few things before a proper merge can be done.

File: modular_zubbers/code/game/objects/items/holy_weapons.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	desc = "A staff used by traditional bishops and popes."
+	icon = 'modular_zubbers/icons/obj/items_and_weapons.dmi'
+	icon_state = "papal_staff"
+	inhand_icon_state = "papal_staff"
+	lefthand_file = 'modular_zubbers/icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'modular_zubbers/icons/mob/inhands/weapons/melee_righthand.dmi'
+	w_class = WEIGHT_CLASS_BULKY
+	attack_verb_continuous = list("smacks", "strucks", "cracks", "beats", "purifies")
+	attack_verb_simple = list("smack", "struck", "crack", "beat", "purify")
+	slot_flags = 0
```
Comment (Cyprex) [inline modular_zubbers/code/game/objects/items/holy_weapons.dm:12]: Can be removed again once I do some CI changes.

Comment (WoolyAypa) [reply modular_zubbers/code/game/objects/items/holy_weapons.dm:12]: I'm not sure what you mean by that, sorry! Should I remove these lines?

Comment (Cyprex) [reply modular_zubbers/code/game/objects/items/holy_weapons.dm:12]: Try removing the slot_flags line now, I pushed a fix for CI.

File: modular_zubbers/code/modules/clothing/head/wig.dm [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/obj/item/clothing/head/wig
+	var/obj/item/clothing/head/head_accessory
+	var/mutable_appearance/head_accessory_MA
+
+	var/overlay_on = FALSE
+
+	var/mob/living/carbon/holder
+	var/item_path
+
+/obj/item/clothing/head/wig/attackby(obj/item/I, mob/living/user)
```
Comment (Cyprex) [inline modular_zubbers/code/modules/clothing/head/wig.dm:10]:
```suggestion
/obj/item/clothing/head/wig/attackby(obj/item/attachment, mob/living/user)
```

Comment (Cyprex) [reply modular_zubbers/code/modules/clothing/head/wig.dm:10]: Or accessory, or to_be_attached, or something like that

File: modular_zubbers/code/modules/clothing/head/wig.dm [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	. = ..()
+	//Checks if the item is in the list of items available
+	for(var/type in HEAD_ACCESSORIES_PATHS)	
+		if(istype(I, type))
+			item_path = type
+			break
```
Comment (Cyprex) [inline modular_zubbers/code/modules/clothing/head/wig.dm:16]: This is fine but if you want you do something like this too
```
if(!is_path_in_list(I.type, HEAD_ACCESSORIES_PATHS)
    break
item_path = I.type
```

File: modular_zubbers/code/modules/clothing/head/wig.dm [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+	if(holder)
+		holder.add_overlay(head_accessory_MA)
+		overlay_on = TRUE
+
+	head_accessory = I
+
+/obj/item/clothing/head/wig/attack_hand_secondary(mob/user)
+	..()
+	. = SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```
Comment (Cyprex) [inline modular_zubbers/code/modules/clothing/head/wig.dm:38]: Are you sure this is correct & what lead you to this?

**Review by Cyprex — CHANGES_REQUESTED** [review]
See above.

In addition to this I'm curious if you couldn't just add the overlay to the item instead(greatly simplifying the code in the process or otherwise modify the items appearance.

File: modular_zubbers/code/modules/clothing/head/wig.dm (hunk 3 above)
Comment (Cyprex) [reply modular_zubbers/code/modules/clothing/head/wig.dm:16]: Actually I'm unsure how this handles subtypes!

Comment (WoolyAypa) [reply modular_zubbers/code/modules/clothing/head/wig.dm:16]: Yeahh.. Don't think there's a way of doing that without iterations

File: modular_zubbers/code/modules/clothing/head/wig.dm (hunk 2 above)
Comment (WoolyAypa) [reply modular_zubbers/code/modules/clothing/head/wig.dm:10]: Uhm, sure!

File: modular_zubbers/code/modules/clothing/head/wig.dm (hunk 4 above)
Comment (WoolyAypa) [reply modular_zubbers/code/modules/clothing/head/wig.dm:38]: Uh well, here I go through the parent conditions and then just cancel the attack chain...
Is there anything wrong with that snippet?

Comment (Cyprex) [reply modular_zubbers/code/modules/clothing/head/wig.dm:38]: Sure, sounds good. 
Might need a
```
	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
		return```
Which is a pattern that shows up occassionally when searching, I can't tell, would have to ask coderbus. But this is fine to merge as is.

**Comment by Cyprex** [comment]
Think you could merge master into your branch? Should hopefully fix the CI failures...

**Comment by WoolyAypa** [comment]
???

**Comment by projectkepler-ru** [comment]
This PR is not usable at the moment, please fix as soon as possible

**Comment by projectkepler-ru** [comment]
Fix your merge conflict then I'll merge it

**Comment by projectkepler-ru** [comment]
@WoolyAypa 
Worn icon are giving off issues, confirm is this working on your locale?

**Comment by Cyprex** [comment]
You forgot to include the .dme

**Pull request merged**